src/PusherNotificationDriver.php

```diff
@@ -11,36 +11,32 @@
 
 use Flarum\Notification\Blueprint\BlueprintInterface;
 use Flarum\Notification\Driver\NotificationDriverInterface;
-use Pusher;
+use Illuminate\Contracts\Queue\Queue;
 
 class PusherNotificationDriver implements NotificationDriverInterface
 {
     /**
-     * @var Pusher
+     * @var Queue
      */
-    protected $pusher;
+    protected $queue;
 
-    public function __construct(Pusher $pusher)
+    public function __construct(Queue $queue)
     {
-        $this->pusher = $pusher;
+        $this->queue = $queue;
     }
 
     /**
      * {@inheritDoc}
      */
     public function send(BlueprintInterface $blueprint, array $users): void
     {
-        foreach ($users as $user) {
-            if ($user->shouldAlert($blueprint::getType())) {
-                $this->pusher->trigger('private-user'.$user->id, 'notification', null);
-            }
-        }
+        $this->queue->push(new SendPusherNotificationsJob($blueprint, $users));
```

Comment:
Perhaps we should only do this if users has length like with Email and Alerts?

Reply:
ah yes you're right, no need to queue a job that won't do anything.